src/zarr/storage/_common.py

```diff
@@ -54,59 +62,90 @@ def __init__(self, store: Store, path: str = "") -> None:
     def read_only(self) -> bool:
         return self.store.read_only
 
+    @classmethod
+    async def _create_open_instance(cls, store: Store, path: str) -> Self:
+        """Helper to create and return a StorePath instance."""
+        await store._ensure_open()
+        return cls(store, path)
+
     @classmethod
     async def open(cls, store: Store, path: str, mode: AccessModeLiteral | None = None) -> Self:
         """
         Open StorePath based on the provided mode.
 
-        * If the mode is 'w-' and the StorePath contains keys, raise a FileExistsError.
-        * If the mode is 'w', delete all keys nested within the StorePath
-        * If the mode is 'a', 'r', or 'r+', do nothing
+        * If the mode is None, return an opened version of the store with no changes.
+        * If the mode is 'r+', 'w-', 'w', or 'a' and the store is read-only, raise a ValueError.
```

Comment:
Given we can go one way (make a store read-only), is there a reason not to go the other way (make a store writeable) in this case, instead of raising an error?

Reply:
I viewed only going from writable -> readable as an optimization for safety because it seems more likely that someone could make a mistake. I haven't seen the pattern of creating a (theoretically) writeable array from a read-only store as much as the opposite case, so could we see if there's a request for that functionality rather than adding it to this PR?

Reply:
👍 sounds good